openedx/features/enterprise_support/middleware.py

```diff
@@ -28,3 +34,33 @@ def process_request(self, request):
         """
         if 'enterprise_customer' not in request.session and request.user.is_authenticated:
             request.session['enterprise_customer'] = api.enterprise_customer_for_request(request)
+
+
+def set_learner_active_enterprise(user=None, backend=None, *args, **kwargs):
```

Comment:
we are removing this middleware so please put this method in enterprise_support and decorate it with `enterprise_is_enabled` decorator.